lib/Test.pm6

```diff
@@ -611,11 +611,21 @@ sub throws-like($code, $ex_type, $reason?, *%matcher) is export {
                 if $type_ok {
                     for %matcher.kv -> $k, $v {
                         my $got is default(Nil) = $_."$k"();
-                        my $ok = $got ~~ $v,;
-                        ok $ok, ".$k matches $v.gist()";
-                        unless $ok {
-                            _diag "Expected: " ~ ($v ~~ Str ?? $v !! $v.perl)
-                              ~ "\nGot:      $got";
+                        # Match strings and Regex as regex, everything else gets a direct comparison.
+                        if (nqp::eqaddr($v.WHAT, Str) or nqp::eqaddr($v.WHAT, Regex)) {
```

Comment:
These are only style comments, not about the change in logic.

Core code doesn't usually use parens about the conditional of an `if`.
Unless performance is a concern, this might be simpler as `$v ~~ Str|Regex`;